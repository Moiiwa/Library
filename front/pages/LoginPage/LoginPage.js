import React from 'react'

import {login} from '../../api/authorization'

import './LoginPage.css'
import {Link} from "react-router-dom";
import {history} from "../../helpers/history";

class LoginPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            user: {
                username: '',
                password: ''
            },
            submitted: false
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        const {name, value} = event.target;
        const {user} = this.state;
        this.setState({
            user: {
                ...user,
                [name]: value
            }
        });
    }

    handleSubmit = async (event) => {
        event.preventDefault();
        this.setState({submitted: true});
        if (this.state.user.username && this.state.user.password) {
            const response = await login(this.state.user.username, this.state.user.password)
        }
    }

    render() {

        const {user, submitted} = this.state;

        return (
            <div className="col-md-6 col-md-offset-3 jumbotron">
                <h1 className="display-4">Login</h1>

                <form name="form" id="submit" onSubmit={this.handleSubmit}>

                    <div className={'form-group' + (submitted && !user.username ? ' has-error' : '')}>
                        <label htmlFor="username">Username</label>
                        <input type="text" className="form-control" name="username" value={user.username}
                               onChange={this.handleChange}/>
                        {submitted && !user.username &&
                        <div className="help-block">Username is required</div>
                        }
                    </div>

                    <div className={'form-group' + (submitted && !user.password ? ' has-error' : '')}>
                        <label htmlFor="password">Password</label>
                        <input type="password" className="form-control" name="password" value={user.password}
                               onChange={this.handleChange}/>
                        {submitted && !user.password &&
                        <div className="help-block">Password is required</div>
                        }
                    </div>

                    <div className="form-group">
                        <button className="btn btn-primary">Log In</button>
                        <Link className="btn btn-link" to={"/registration"}>Register</Link>
                    </div>
                </form>

            </div>
        )
    }
}


export default LoginPage;