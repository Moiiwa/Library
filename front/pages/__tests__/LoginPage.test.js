import React from 'react';
import LoginPage from '../LoginPage/LoginPage';
import toJson from 'enzyme-to-json';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('LoginPage', () => {
    let wrapper;
    let mockSubmit;

    beforeEach(() => {
        mockSubmit = jest.fn();
        wrapper = shallow(<LoginPage submit={mockSubmit} />);
    });

    it('renders correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("handle change", () => {
        const mockEvent = {
            target: {
                name: "username",
                value: "test"
            }
        };

        const expected = {
            user: {
                username: "test",
                password: "",
            },
            "submitted": false,
        };

        wrapper.instance().handleChange(mockEvent);
        expect(wrapper.state()).toEqual(expected)
    });

    it('handle submit', () => {
        const fakeEvent = { preventDefault: () => console.log('preventDefault') };

        const instance = wrapper.instance()
        const handleSubmitMock = jest.spyOn(instance, 'handleSubmit')
        instance.forceUpdate();

        const but = wrapper.find('#submit')
        but.simulate('submit', fakeEvent);
        expect(handleSubmitMock).toHaveBeenCalled()
    });

});