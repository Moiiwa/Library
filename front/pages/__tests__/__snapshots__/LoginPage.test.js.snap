// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders correctly 1`] = `
<div
  className="col-md-6 col-md-offset-3 jumbotron"
>
  <h1
    className="display-4"
  >
    Login
  </h1>
  <form
    id="submit"
    name="form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        className="form-control"
        name="username"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <button
        className="btn btn-primary"
      >
        Log In
      </button>
      <Link
        className="btn btn-link"
        replace={false}
        to="/registration"
      >
        Register
      </Link>
    </div>
  </form>
</div>
`;
